### DISCUSSION TITLE
IWYU is noise in our commit history

### DISCUSSION THREAD

**Discussion by pierricgimmig** [google/orbit#4806] [General]
Could we reduce the frequency of IWYU? 

Also, it's weird to me that we have consecutive commits of IWYU, sometimes a commit undoes what the previous commit.

#4789 
#4788 

#4785
#4784

**Comment by florian-kuebler** [comment]
>Also, it's weird to me that we have consecutive commits of IWYU, sometimes a commit undoes what the previous commit.

That's clearly a bug. Probably we need to adjust the mappings a bit. I'll try to fix it.

>Could we reduce the frequency of IWYU?

Well, currently, we perform the action after each commit to `main`, unless there is an open IWYU PR.
We could change the trigger, to a schedule, and e.g. run it once a week. This would be a bit of a resource waste if development here on the repo slows down. But, I don't have a strong opinion on this.

Ideally, the check would run on each PR, but it is simply to slow for that.

**Comment by florian-kuebler** [comment]
in your given examples, i can't see IWYU undo previous changes. What am I missing?
